src/infrastructure/persistence/PsqlUserRepository.ts

```diff
@@ -69,19 +83,111 @@ export class PsqlUserRepository implements UserRepository {
       return null;
     }
-    return extractUser(userRow);
+    return this.extractUserAndPopulateRoles(userRow);
+  }
+
+  private async saveRoleAssignment(roleAssignment: RoleAssignmentAction) {
+    await this.db('role_to_user_assignment').insert({
+      id: roleAssignment.id.value,
+      user_id: roleAssignment.assignedTo.id.value,
+      role_name: roleAssignment.assignedResource.name,
+      actor: roleAssignment.actor.value,
+      creation_time: roleAssignment.creationTime,
+      order: roleAssignment.order,
+      action_type: roleAssignment.actionType.toString(),
+    });
+  }
+
+  private async extractUserAndPopulateRoles(userRow: any) {
+    const user = new User(
+      new UserId(userRow.id),
+      new Email(userRow.email),
+      fromDbProfile(userRow.profile as DbProfile | undefined),
+      fromDbAddress(userRow.address as DbAddress | undefined),
+      userRow.creationTime,
+      userRow.modificationTime
+    );
+    const roleAssignmentActions = await this.getRoleAssignments(user);
+    Reflect.set(
+      user.roleAssignments,
+      'assignmentActions',
+      roleAssignmentActions
+    );
+    Reflect.set(user.roleAssignments, 'newAssignmentActions', []);
+    return user;
   }
-}
 
-function extractUser(userRow: any) {
-  return new User(
-    new UserId(userRow.id),
-    new Email(userRow.email),
-    fromDbProfile(userRow.profile as DbProfile | undefined),
-    fromDbAddress(userRow.address as DbAddress | undefined),
-    userRow.creationTime,
-    userRow.modificationTime
-  );
+  private async getRoleAssignments(
+    user: User
+  ): Promise<RoleAssignmentAction[]> {
+    const roleAssignmentRows: any[] = await this.db(
+      'role_to_user_assignment as rua'
+    )
+      .select([
+        'rua.id as rua_id',
+        'rua.creation_time as rua_creation_time',
+        'rua.action_type as rua_action_type',
+        'rua.actor as rua_actor',
+        'rua.order as rua_order',
+        'r.name as role_name',
+        'r.creation_time as role_creation_time',
+        'pra.id as pra_id',
+        'pra.creation_time as pra_creation_time',
+        'pra.action_type as pra_action_type',
+        'pra.actor as pra_actor',
+        'pra.order as pra_order',
+        'p.name as permission_name',
+        'p.creation_time as permission_creation_time',
+      ])
+      .leftJoin('role as r', 'r.name', 'rua.role_name')
+      .leftJoin(
+        'permission_to_role_assignment as pra',
+        'r.name',
+        'pra.role_name'
+      )
+      .leftJoin('permission as p', 'pra.permission_name', 'p.name')
+      .where('rua.user_id', '=', user.id.value);
+
+    const groupedByAssignmentId = this.groupByAssignmentId(roleAssignmentRows);
+    return this.mapToRoleAssignmentActions(groupedByAssignmentId, user);
+  }
+
+  private groupByAssignmentId(roleAssignmentRows: any[]) {
+    const groupedByAssignmentId = new Map<string, any[]>();
+    roleAssignmentRows
+      .filter((row) => row.role_name && row.role_creation_time)
```

Comment:
Don't need parenthesis again `(row) => `

Reply:
We don't need this but this is how prettier is defined. try to remove it and you'll see it'll add it back. :)